File: web/src/lib/models/cacheTreeModels.ts
export type TimestampMetadata = {
  date: string;
  time: string;
};

export type PreviewMetadata = {
  resultTimestamp: TimestampMetadata;
  operator: string;
  xColumnNames: string[];
  yColumnNames: string[];
  totalRows: number;
};

export type ColumnNode = {
  rows: number[];
};

export type YColumnMetadata = {
  samples: number;
  date: string;
  unitLabel: string;
  xDimension: string;
};

File: pkg/staticfire/cache_tree_models.go
package staticfire

type TimestampMetadata struct {
	Date string `json:"date"`
	Time string `json:"time"`
}

// Represents metadata for previewing the results of a static fire.
type PreviewMetadata struct {
	ResultTimestamp TimestampMetadata `json:"resultTimestamp"`
	Operator        string            `json:"operator"`
	XColumnNames    []string          `json:"xColumnNames"`
	YColumnNames    []string          `json:"yColumnNames"`
	TotalRows       int               `json:"totalRows"`
}

// Represents a column node in the cache tree.
type ColumnNode struct {
	Rows []float64 `json:"rows"`
}

// Represents metadata for the Y column in the cache tree.
type YColumnMetadata struct {
	Samples    int    `json:"samples"`
	Date       string `json:"date"`
	UnitLabel  string `json:"unitLabel"`
	XDimension string `json:"xDimension"`
}

// Represents the file structure to create a cache tree.
type CacheTree struct {
	PreviewMetadata PreviewMetadata
	YColumnMetadata []YColumnMetadata // Matches the length of YRows
	XColumnNodes    []ColumnNode
	YColumnNodes    []ColumnNode
}
